src/common/meta/src/ddl/table_meta.rs

```diff
@@ -115,8 +123,31 @@ impl TableMetadataAllocator {
     ) -> Result<TableRouteValue> {
         let regions = task.partitions.len();
 
-        let table_route = if task.create_table.engine == METRIC_ENGINE {
-            TableRouteValue::Logical(LogicalTableRouteValue {})
+        let table_route = if task.create_table.engine == METRIC_ENGINE
+            && let Some(physical_table_name) = task
+                .create_table
+                .table_options
+                .get(LOGICAL_TABLE_METADATA_KEY)
```

Comment:
What if the `table_options` is missing `LOGICAL_TABLE_METADATA_KEY`? Should we change to following: 
```rust 
let table_route = if task.create_table.engine == METRIC_ENGINE {
            let physical_table_name = task
                .create_table
                .table_options
                .get(LOGICAL_TABLE_METADATA_KEY)
                .context(error::UnexpectedSnafu {})?;
        
```